src/infrastructure/ui/pages/root-page/hooks.ts

```diff
@@ -0,0 +1,91 @@
+            navigate(route);
+            setActiveScreen(route);
+        };
+    };
+
+    const [footerY, setFooterY] = useState(750);
+    const footerRef = useRef<View>(null);
+
+    footerRef.current?.measure(
+        (_x: number, _y: number, _weight: number, _height: number, _pageX: number, pageY: number) => {
+            setFooterY(pageY);
+        }
+    );
+
+    const animatedHistoricalIconStyle = StyleSheet.create({ icon: {} });
+    const animatedScanIconStyle = StyleSheet.create({ icon: {} });
+    const animatedHomeIconStyle = StyleSheet.create({ icon: {} });
+    const animatedRecipeIconStyle = StyleSheet.create({ icon: {} });
+    const animatedGameIconStyle = StyleSheet.create({ icon: {} });
+    const animatedSlideBar = StyleSheet.create({ slideBar: {} });
+
+    if (activeScreen == PagesEnum.HistoricalPage) {
+        animatedHistoricalIconStyle.icon = { marginBottom: 20 };
+        animatedSlideBar.slideBar = { left: 12 };
+    } else if (activeScreen == PagesEnum.ScanPage) {
+        animatedScanIconStyle.icon = { marginBottom: 20 };
+        animatedSlideBar.slideBar = { left: 12 + Dimensions.get('screen').width / 5 };
+    } else if (activeScreen == PagesEnum.RecipePage) {
+        animatedRecipeIconStyle.icon = { marginBottom: 20 };
+        animatedSlideBar.slideBar = { left: 12 + (3 * Dimensions.get('screen').width) / 5 };
+    } else if (activeScreen == PagesEnum.GamePage) {
+        animatedGameIconStyle.icon = { marginBottom: 20 };
+        animatedSlideBar.slideBar = { left: 12 + (4 * Dimensions.get('screen').width) / 5 };
+    } else {
+        animatedHomeIconStyle.icon = { marginBottom: 25 };
+        animatedSlideBar.slideBar = { left: 12 + (2 * Dimensions.get('screen').width) / 5 };
+    }
+
+    const historicalPageAsset = require('~/domain/entities/assets/icon/footer-icon/icon-historical-page.svg');
+    const scanPageAsset = require('~/domain/entities/assets/icon/footer-icon/icon-scan-page.svg');
+    const homePageAsset = require('~/domain/entities/assets/icon/footer-icon/icon-small-white-logo.svg');
+    const recipePageAsset = require('~/domain/entities/assets/icon/footer-icon/icon-recipe-page.svg');
+    const gamePageAsset = require('~/domain/entities/assets/icon/footer-icon/icon-game-page.svg');
+
+    const newHeight = 50;
+    const newWidth = 50;
+
+    const returnObject = {
+        animatedGameIconStyle,
+        animatedHistoricalIconStyle,
+        animatedHomeIconStyle,
+        animatedRecipeIconStyle,
+        animatedScanIconStyle,
+        animatedSlideBar,
+        footerRef,
+        footerY,
+        gamePageAsset,
+        historicalPageAsset,
+        homePageAsset,
+        recipePageAsset,
+        scanPageAsset,
+        newHeight,
+        newWidth,
+        onPressIcon,
+        Stack
+    };
+
+    try {
+        oldNavigation.reset({ index: 0, root: [{ name: PagesEnum.RootPage }] });
+    } catch (e) {
+        return returnObject;
+    }
+
+    return returnObject;
```

Comment:
Du coup ici le catch qui fait un return avec un return en dehors ne sert pas à grand chose du coup on pourrait remplacer le catch par un finally:

```suggestion
    } finally (e) {
         return returnObject;
    }
};

Reply:
Ca me met une erreur de any avec le e ou une erreur de unsafe return donc pas ouf